src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs

```diff
@@ -56,11 +63,13 @@ public SqlConnection(string databasePath)
 
             _handle = handle;
             raw.sqlite3_busy_timeout(_handle, (int)TimeSpan.FromMinutes(1).TotalMilliseconds);
+
+            _successfullyCreated = true;
```

Comment:
Other approaches were: 

1. split into Construction+initialization phase.  That had more downstream effects.
2. have  try/catch here and call GC.SuppressFinalization.  Seemed oogier than this approach.

i tried to keep this simple, though i don't love how the code is.